app/models/submission_collector.rb

```diff
@@ -274,18 +274,20 @@ def uncollect_submissions(assignment)
     old_submissions = submissions.where(submission_version_used: true)
     ActiveRecord::Base.transaction do
       old_submissions.each do |submission|
-        version = submission.submission_version
         grouping = submission.grouping
-        if version == 1
-          grouping.assign_attributes(is_collected: false)
-        else
-          prev_rev = submissions.where(submission_version: version - 1,
-                                       grouping_id: grouping.id).first
-          prev_rev.update_attributes(submission_version_used: true)
-        end
-        submission.update_attributes(submission_version_used: false)
-        grouping.update_attributes(grouping_queue_id: nil)
+        grouping.update_attributes(grouping_queue_id: nil, is_collected: false)
+        # version = submission.submission_version
+        # grouping = submission.grouping
+        # if version == 1
+        #   grouping.assign_attributes(is_collected: false)
+        # else
+        #   prev_rev = submissions.where(submission_version: version - 1,
+        #                                grouping_id: grouping.id).first
+        #   prev_rev.update_attributes(submission_version_used: true)
```

Comment:
This doesn't seem to have the correct behaviour.  If I collect all submissions, do some marking, and then uncollect, the previous collection is not restored. submission_version_used for the previously collected version is still false.

(I'm trying to debug it a bit and simply uncommenting this block doesn't seem to fix it.)